### PR TITLE
Add geopmadmin tool

### PR DESCRIPTION
**Pull request by cmcantalupo**

### CODE REVIEW

File: ronn/geopmadmin.1.ronn [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+[//]: # (LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,)
+[//]: # (DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY)
+[//]: # (THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT)
+[//]: # ((INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE)
+[//]: # (OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.)
+
+## SYNOPSIS
+
+`geopmadmin` [`--config-default`|`--config-override`|`--msr-whitelist`]
+
```
Comment (dianarg) [inline ronn/geopmadmin.1.ronn:37]: help and version?

File: ronn/geopmadmin.1.ronn [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+## DESCRIPTION
+
+Used by system administrators to check the validity of the GEOPM
+configuration on the system.  When run with no arguments it will parse
+the default and override configuration files on the system and print
+out the values defined by these files.  This command line tool can be
+used to discover the location of the geopm configuration files
+the set default and override values for the GEOPM runtime.  The tool
```
Comment (dianarg) [inline ronn/geopmadmin.1.ronn]: the set -> that set

File: ronn/geopmadmin.1.ronn [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+can also generate a whitelist for the msr-safe tool that meets the
+GEOPM requirements.
```
Comment (dianarg) [inline ronn/geopmadmin.1.ronn]: Might also want back link(s) to the part of the README that discusses these set up steps.

File: ronn/geopmadmin.1.ronn [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  * `-w`, `--msr-whitelist`:
+    Print the minimum required whitelist for the msr-safe driver to
+    enable all of the GEOPM features.
+
+## COPYRIGHT
+Copyright (c) 2015, 2016, 2017, 2018, 2019, Intel Corporation. All rights reserved.
+
+## SEE ALSO
+**geopm(7)**,
+**geopmagent(1)**
```
Comment (dianarg) [inline ronn/geopmadmin.1.ronn:93]: geopmlaunch(1) ?

**Review by dianarg — CHANGES_REQUESTED** [review]
Any spec file changes needed?  Should this go into beta for now until the next release, or would it be better to move to beta only if it's not part of the release?

File: ronn/geopmadmin.1.ronn [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+[//]: # (OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.)
+
+## SYNOPSIS
+
+`geopmadmin` [`--config-default`|`--config-override`|`--msr-whitelist`]
+
+## DESCRIPTION
+
+Used by system administrators to check the validity of the GEOPM
+configuration on the system.  When run with no arguments it will parse
```
Comment (bakerbrandond) [inline ronn/geopmadmin.1.ronn]: output is assumed json.  should we be explicit here?

Comment (cmcantalupo) [reply ronn/geopmadmin.1.ronn]: I was thinking more human readable:

GEOPM_AGENT: energy_efficient (default)
GEOPM_POLICY: /etc/geopm/policy.json (default)
GEOPM_PLUGIN_PATH: /opt/admin/geopm/plugin (override)

Comment (bakerbrandond) [reply ronn/geopmadmin.1.ronn]: can I request an option for json so I can dump the output of the tool directly into a json object in python for the testing?

**Review by dianarg — APPROVED** [review]

**Pull request merged**